src/components/ProductListing.vue

```diff
@@ -27,26 +19,50 @@
 </template>
 <script>
 import axios from 'axios'
+import NavBar from './NavBar.vue'
+import ProductCard from './ProductCard.vue'
 export default {
-  name: "ProductCard",
+  name: "ProductListing",
   data()
   {
       return{
           productList:[],
-          current_page: 1,
-          product_per_page: 20,
+          currentPage: 0,
+          perPage: 12,
+          paginatedItems:[],
+          page_count: 0
       }
   },
+  components: {
+    NavBar,
+    ProductCard,
+  },
   methods:{
-      calculateDiscoutedAmount(product){
-          return (product.price-((product.price*product.discountPercentage)/100)).toFixed(2)
-      }
+
+    displayProductList(){
+        let start = this.perPage * this.currentPage;
+        let end = start + this.perPage;
+        this.paginatedItems =  this.productList.slice(start, end);
+    },
+
+    setupPagination(){
+        this.page_count = Math.ceil(this.productList.length / this.perPage);
+    },
+
+    paginationButton(page){
+        this.currentPage=page;
+        this.displayProductList()
+    },
 
   },
   created()
   {
       axios.get('https://dummyjson.com/products')
-      .then((response) => this.productList = response.data.products)
+      .then(({data}) => {
```

Comment:
we can try seting up axios globally. prefer this docs https://axios-http.com/docs/config_defaults

Reply:
done